include/util/static_rtree.hpp

```diff
@@ -166,15 +167,245 @@ class StaticRTree
     StaticRTree(const StaticRTree &) = delete;
     StaticRTree &operator=(const StaticRTree &) = delete;
 
-    // Construct a packed Hilbert-R-Tree with Kamel-Faloutsos algorithm [1]
+    enum PackingMethod
+    {
+        Hilbert,
+        STR
+    };
+
     explicit StaticRTree(const std::vector<EdgeDataT> &input_data_vector,
                          const std::string &tree_node_filename,
                          const std::string &leaf_node_filename,
-                         const Vector<Coordinate> &coordinate_list)
+                         const Vector<Coordinate> &coordinate_list,
+                         const PackingMethod packing_method = Hilbert)
         : m_coordinate_list(coordinate_list)
     {
+        util::Log() << "Starting rtree";
+        switch (packing_method)
```

Comment:
This is good for the algorithm selection at run-time, but packing method can be generalized out and selected at compile-time using a packing policy